src/utils.ts

```diff
@@ -35,8 +44,126 @@ export class Utils {
+        }
+    }
 
-    public static async addCliToPath() {
+    /**
+     * Gathers JFrog's credentials from environment variables and delivers them in a JfrogCredentials structure
+     * @private
+     */
+    public static collectJfrogCredentialsFromEnvVars(): JfrogCredentials {
+        let jfrogCredentials : JfrogCredentials = {} as JfrogCredentials;
+        console.log("Searching for JF_URL")
+        if (process.env.JF_URL) {
+            console.log("JF_URL found")
+            jfrogCredentials.jfrogUrl = process.env.JF_URL;
+        }
+
+        console.log("Searching for JF_ACCESS_TOKEN, JF_USER and JF_PASSWORD")
+        if (process.env.JF_ACCESS_TOKEN) {
+            console.log("JF_ACCESS_TOKEN found")
+            jfrogCredentials.accessToken = process.env.JF_ACCESS_TOKEN
+        }
+
+        if (process.env.JF_USER && process.env.JF_PASSWORD) {
+            console.log("JF_USER and JF_PASSWORD found")
+            jfrogCredentials.username = process.env.JF_USER;
+            jfrogCredentials.password = process.env.JF_PASSWORD;
+        }
+        return jfrogCredentials
+    }
+
+    /**
+     * Exchanges JWT with a valid access token
+     * @param jfrogCredentials existing JFrog credentials - url, access token, username + password
+     * @param jsonWebToken JWT achieved from GitHub JWT provider
+     * @private
+     */
+    private static async getAccessTokenFromJWT(jfrogCredentials: JfrogCredentials, jsonWebToken: string): Promise<JfrogCredentials> {
+        // @ts-ignore : If we've reached this stage, the jfrogCredentials.jfrogUrl field should hold a non-empty value obtained from process.env.JF_URL
+        const exchangeUrl : string = jfrogCredentials.jfrogUrl.replace(/\/$/, '') + "/access/api/v1/oidc/token"
+        console.log("Exchanging JSON web token with access token")
+
+        const provider_name: string = core.getInput(Utils.OIDC_INTEGRATION_PROVIDER_NAME, { required: true });
+        const httpClient : HttpClient = new HttpClient()
+        try {
+            const data: string = `{
+                "grant_type": "urn:ietf:params:oauth:grant-type:token-exchange",
+                "subject_token_type": "urn:ietf:params:oauth:token-type:id_token",
+                "subject_token": "${jsonWebToken}",
+                "provider_name": "${provider_name}"
+            }`;
+
+            const additionalHeaders: OutgoingHttpHeaders = {
+                'Content-Type': 'application/json',
+            };
+
+            const response: HttpClientResponse = await httpClient.post(exchangeUrl, data, additionalHeaders)
+            const responseString : string = await response.readBody()
+            const responseJson : TokenExchangeResponseData = JSON.parse(responseString)
+            jfrogCredentials.accessToken = responseJson.access_token;
+        } catch (error : any) {
+            throw new Error(`POST REST command failed with error ${error.message}`)
+        }
```

Comment:
There is already a try-and-catch outside of this function
```suggestion
```

Reply:
but at this point we can give more specific info about the failure. isn't it better?